src/offset/local/windows.rs

```diff
@@ -144,39 +143,54 @@ impl TzInfo {
         Some(TzInfo {
             std_offset: FixedOffset::west_opt((tz_info.Bias + tz_info.StandardBias) * 60)?,
             dst_offset: FixedOffset::west_opt((tz_info.Bias + tz_info.DaylightBias) * 60)?,
-            std_transition: system_time_from_naive_date_time(tz_info.StandardDate, year),
-            dst_transition: system_time_from_naive_date_time(tz_info.DaylightDate, year),
+            std_transition: naive_date_time_from_system_time(tz_info.StandardDate, year).ok()?,
+            dst_transition: naive_date_time_from_system_time(tz_info.DaylightDate, year).ok()?,
         })
     }
 }
 
-fn system_time_from_naive_date_time(st: SYSTEMTIME, year: i32) -> Option<NaiveDateTime> {
+fn naive_date_time_from_system_time(
+    st: SYSTEMTIME,
+    year: i32,
+) -> Result<Option<NaiveDateTime>, ()> {
     if st.wYear == 0 && st.wMonth == 0 {
-        return None; // No DST transitions for this year in this timezone.
+        return Ok(None); // No DST transitions for this year in this timezone.
```

Comment:
Why is this? It doesn't intuitively make sense to me that this needs to return `Result<Option<_>, _>` instead of just `Result`.

Reply:
In a way `SYSTEMTIME` is an `Option`-like type. If it is `0` it means there is no transition date that year. If it is non-zero, it can be a concrete date or a rule.

We can resolve the rule to a concrete date within this function, so besides returning `NaiveDateTime` we only need to encode two other cases: `SYSTEMTIME` was `None`, or there was an error.

Reply:
The function is called `naive_date_time_from_system_time()`, which makes it non-obvious to me why we would need to return `None` in case there are no DST transitions.

Reply:
I'll add a doc comment and think up a better name :smile:.

Reply:
Maybe the name is fine -- the context of `SYSTEMTIME` being `Option`-like wasn't clear to me; if we can resolve that with documentation it would okay.